File: backend/src/main/java/com/chicago/labs/domain/Pair.java
package com.chicago.labs.domain;

public class Pair {
    private Human first;
    private Human second;

    public Pair() {
    }

    public Pair(final Human first, final Human second) {
        this.first = first;
        this.second = second;
    }

    public Human getFirst() {
        return first;
    }

    public Human getSecond() {
        return second;
    }

    @Override
    public String toString() {
        return "Pair{" +
                "first=" + first +
                ", second=" + second +
                '}';
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        final Pair pair = (Pair) o;

        if (first != null ? !first.equals(pair.first) : pair.first != null) return false;
        return second != null ? second.equals(pair.second) : pair.second == null;
    }

    @Override
    public int hashCode() {
        int result = first != null ? first.hashCode() : 0;
        result = 31 * result + (second != null ? second.hashCode() : 0);
        return result;
    }
}

File: backend/src/main/kotlin/com/chicago/labs/domain/Pair.kt
package com.chicago.labs.domain

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.PersistenceConstructor
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Pair
@PersistenceConstructor constructor(
        @Id val id: ObjectId? = null,
        var first: Human? = null,
        var second: Human? = null
)